mlir/include/mlir/Dialect/AMDGPU/IR/AMDGPU.td

```diff
@@ -35,6 +35,11 @@ def AMDGPU_Dialect : Dialect {
   let useDefaultAttributePrinterParser = 1;
 }
 
+def AnyIntegerOrFloat : AnyTypeOf<[AnySignlessInteger, AnyFloat], "Integer or Float">;
```

Comment:
I don't think these are restricted to signless values.

And also ... really, anything with a bitwidth is allowed.

I think we realistically want whatever the checks are for "can be in a vector" here

Reply:
I copypasted this check from the `gpu.shuffle`, non-signless ints are not very useful in such a low level dialect and all arith/llvm ops (which I will use for repacking) expecting signless anyway, so I won't be able to do anything with them.